uint32 state
uint32 STOPPED = 0
uint32 RUNNING = 1
uint32 ACTIVE = 2
